src/config/config.rs

```diff
@@ -350,33 +347,22 @@ impl Config {
       Source::Yml => Ok(Config::from_yaml(schema)?),
     }
   }
-
   pub fn n_plus_one(&self) -> Vec<Vec<(String, String)>> {
     super::n_plus_one::n_plus_one(self)
   }
 
-  pub async fn from_file_paths(file_paths: std::slice::Iter<'_, String>) -> Result<Config> {
-    let mut config = Config::default();
-    let futures: Vec<_> = file_paths
-      .map(|file_path| async move {
-        let source = Source::detect(file_path)?;
-        let mut f = File::open(file_path).await?;
-        let mut buffer = Vec::new();
-        f.read_to_end(&mut buffer).await?;
-
-        let server_sdl = String::from_utf8(buffer)?;
-        Config::from_source(source, &server_sdl)
-      })
-      .collect();
-
-    for res in join_all(futures).await {
-      match res {
-        Ok(conf) => config = config.clone().merge_right(&conf),
-        Err(e) => return Err(e), // handle error
-      }
+  pub async fn from_file_or_url(file_paths: std::slice::Iter<'_, String>) -> Result<ConfigReader> {
```

Comment:
```suggestion
  pub async fn from_file_or_url(file_paths: std::slice::Iter<'_, String>) -> Result<Config> {
```

Let's return a config for now. Until we know exactly how polling etc. should be implemented.